src/primitives/zcash_note_encryption_domain.rs

```diff
@@ -0,0 +1,295 @@
+//! This module implements `Domain` and `BatchDomain` traits from the `zcash_note_encryption`
+//! crate and contains the common logic for `OrchardVanilla` and `OrchardZSA` flavors.
+
+// Review hint: this file is largely derived from src/note_encryption.rs
+
+use alloc::vec::Vec;
+use blake2b_simd::{Hash, Params};
+use group::ff::PrimeField;
+
+use zcash_note_encryption::{
+    note_bytes::NoteBytes, BatchDomain, Domain, EphemeralKeyBytes, OutPlaintextBytes,
+    OutgoingCipherKey, OUT_PLAINTEXT_SIZE,
+};
+
+use crate::{
+    address::Address,
+    keys::{
+        DiversifiedTransmissionKey, Diversifier, EphemeralPublicKey, EphemeralSecretKey,
+        OutgoingViewingKey, PreparedEphemeralPublicKey, PreparedIncomingViewingKey, SharedSecret,
+    },
+    note::{ExtractedNoteCommitment, Note, RandomSeed, Rho},
+    primitives::{orchard_domain::OrchardDomain, orchard_primitives::OrchardPrimitives},
+    value::{NoteValue, ValueCommitment},
+};
+
+const PRF_OCK_ORCHARD_PERSONALIZATION: &[u8; 16] = b"Zcash_Orchardock";
+
+const NOTE_VERSION_SIZE: usize = 1;
+const NOTE_DIVERSIFIER_SIZE: usize = 11;
+const NOTE_VALUE_SIZE: usize = 8;
+const NOTE_RSEED_SIZE: usize = 32; // rseed (or rcm prior to ZIP 212)
+
+const NOTE_VERSION_OFFSET: usize = 0;
+const NOTE_DIVERSIFIER_OFFSET: usize = NOTE_VERSION_OFFSET + NOTE_VERSION_SIZE;
+const NOTE_VALUE_OFFSET: usize = NOTE_DIVERSIFIER_OFFSET + NOTE_DIVERSIFIER_SIZE;
+const NOTE_RSEED_OFFSET: usize = NOTE_VALUE_OFFSET + NOTE_VALUE_SIZE;
+
+/// The size of a Vanilla compact note.
+pub(super) const COMPACT_NOTE_SIZE_VANILLA: usize =
+    NOTE_VERSION_SIZE + NOTE_DIVERSIFIER_SIZE + NOTE_VALUE_SIZE + NOTE_RSEED_SIZE;
```

Comment:
_From str4d ([r2599745324](https://github.com/zcash/orchard/pull/471#discussion_r2599745324))_

Since we've already done the addition above:
```suggestion
    NOTE_RSEED_OFFSET + NOTE_RSEED_SIZE;
```

Reply:
FIXED